spython/main/parse/parsers/base.py

```diff
@@ -0,0 +1,95 @@
+
+# Copyright (C) 2017-2019 Vanessa Sochat.
+
+# This Source Code Form is subject to the terms of the
+# Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+# with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+from spython.logger import bot
+from spython.utils import read_file
+from ..recipe import Recipe
+
+class ParserBase(object):
+    '''a parser Base is intended to provide helper functions for a parser,
+       namely to read lines in files, and otherwise interact with outputs.
+       Input should be some recipe (text file to describe a container build)
+       and output of parse() is a Recipe (spython.main.parse.recipe.Recipe)
+       object, which can be used to write to file, etc.
+    '''
+
+    lines = []
+
+    def __init__(self, filename=None, load=True):
+        '''a generic recipe parser holds the original file, and provides 
+           shared functions for interacting with files. If the subclass has
+           a parse function defined, we parse the filename
+
+           Parameters
+           ==========
+           recipe: the recipe file to parse.
+
+        '''
+        self.filename = filename
+        self._run_checks()
```

Comment:
To generic name and does more than this. I'd also require a filename in all cases or none at all. Having multiple ways to do things makes things harder to reason about

Reply:
Good thinking.